### PR TITLE
Batch Request for owners of domains

### PR DESCRIPTION
**Pull request by rommex**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,22 @@ package.dependencies.append(
   }
  ```
  
+ ## Batch request of owners
+ In the version 0.1.3 there was introduced a method `batchOwners(domains: _, completion: _ )` that adds additional convenience to query the owners of the array of domains. The domains must be only CNS-compatible (other kind of domains will throw `ResolutionError.methodNotSupported`). The result will return only if all domains are resolved.
+ 
+ ```swift
+ 
+ resolution.batchOwners(domains: ["brad.crypto", "otherbrad.crypto"]) { result in
```
Comment (merenkoff) [inline README.md:82]: maybe better name is owners instead of batchOwners

Comment (merenkoff) [inline README.md:82]:
```suggestion
 resolution.owners(domains: ["brad.crypto", "otherbrad.crypto"]) { result in
```

File: README.md [hunk 2]
```diff
@@ -106,6 +122,8 @@ enum ResolutionError: Error {
     case unspecifiedResolver
     case unknownError(Error)
     case proxyReaderNonInitialized
+    case inconsistenDomainArray
```
Comment (merenkoff) [inline README.md:125]: strange error

File: Sources/UnstoppableDomainsResolution/APIRequest.swift [hunk 3]
```diff
@@ -47,13 +57,15 @@ public struct DefaultNetworkingLayer: NetworkingLayer {
...
+                             httpBody: Data,
+                             completion: @escaping(Result<JsonRpcResponseArray, APIError>) -> Void) {
         do {
             var urlRequest = URLRequest(url: url)
             urlRequest.httpMethod = httpMethod
             urlRequest.addValue(httpHeaderContentType, forHTTPHeaderField: "Content-Type")
-            urlRequest.httpBody = try JSONEncoder().encode(httpBody)
+            urlRequest.httpBody = httpBody
+            
+            let strin = String(data: httpBody, encoding: .utf8)
```
Comment (merenkoff) [inline Sources/UnstoppableDomainsResolution/APIRequest.swift:68]: strange name of variable

File: Sources/UnstoppableDomainsResolution/APIRequest.swift [hunk 4]
```diff
@@ -74,4 +86,35 @@ public struct DefaultNetworkingLayer: NetworkingLayer {
...
+//                    completion(.success(result))
+//                } catch {
+//                    completion(.failure(.decodingError))
+//                }
+//            }
+//            dataTask.resume()
+//        } catch {
+//            completion(.failure(.encodingError))
+//        }
+//    }
```
Comment (merenkoff) [inline Sources/UnstoppableDomainsResolution/APIRequest.swift:119]: please don't provide a commented code for production

File: Sources/UnstoppableDomainsResolution/Helpers/Types.swift [hunk 5]
```diff
@@ -9,5 +9,6 @@
 import Foundation
 
 public typealias StringResultConsumer = (Result<String, ResolutionError>) -> Void
+public typealias StringsArratResultConsumer = (Result<[String?], ResolutionError>) -> Void
```
Comment (merenkoff) [inline Sources/UnstoppableDomainsResolution/Helpers/Types.swift:12]:
```suggestion
public typealias StringsArrayResultConsumer = (Result<[String?], ResolutionError>) -> Void
```

File: Sources/UnstoppableDomainsResolution/Resolution.swift [hunk 6]
```diff
@@ -273,7 +309,15 @@ public class Resolution {
             completion(.failure(.unknownError(error)))
             return
         }
-
+        completion(.failure(catched))
+    }
+    
+    /// Process the 'error'
+    private func catchError(_ error: Error, completion:@escaping StringsArratResultConsumer ) {
```
Comment (merenkoff) [inline Sources/UnstoppableDomainsResolution/Resolution.swift:316]:
```suggestion
    private func catchError(_ error: Error, completion:@escaping StringsArrayResultConsumer ) {
```

File: Sources/UnstoppableDomainsResolution/Resolution.swift [hunk 7]
```diff
@@ -97,6 +97,22 @@ public class Resolution {
         }
     }
 
+    /// Resolves owner addresses of an array of `domain`s
+    /// - Parameter domains: - array of domain names
+    /// - Parameter completion: A callback that resolves `Result`  with an array of `owner address`'s or `Error`
+    public func batchOwners(domains: [String], completion: @escaping StringsArratResultConsumer ) {
```
Comment (merenkoff) [inline Sources/UnstoppableDomainsResolution/Resolution.swift:103]:
```suggestion
    public func batchOwners(domains: [String], completion: @escaping StringsArrayResultConsumer ) {
```

**Review by merenkoff — CHANGES_REQUESTED** [review]
Some naming errors and other comments

File: README.md (hunk 2 above)
Comment (rommex) [reply README.md:125]: This is when domains submitted in one array are supported by different NS, while they should be consistently by one NS

File: README.md (hunk 1 above)
Comment (rommex) [reply README.md:82]: Just 'owners' doesn't reflect the idea of the batch request. It could have been just a sequence of requests...

Comment (rommex) [reply README.md:82]: Just `owners` may mean a sequence of requests. I wanted to stress that this method uses one batch request

**Review by merenkoff — APPROVED** [review]

File: Sources/UnstoppableDomainsResolution/Resolution.swift [hunk 8]
```diff
@@ -97,6 +97,22 @@ public class Resolution {
         }
     }
 
+    /// Resolves owner addresses of an array of `domain`s
+    /// - Parameter domains: - array of domain names
+    /// - Parameter completion: A callback that resolves `Result`  with an array of `owner address`'s or `Error`
```
[the remark this answers is not in the document]
Comment (rommex) [reply Sources/UnstoppableDomainsResolution/Resolution.swift:102]: Currently the resulting array of owners is ordered and aligned with the array of domains. One-to-one consistency

**Pull request merged**